prompt_toolkit/shortcuts/progress_bar/base.py

```diff
@@ -313,6 +317,8 @@ def __init__(self, progress_bar: ProgressBar,
         self.remove_when_done = remove_when_done
         self._done = False
         self.total: Optional[int]
+        for name, label in labels.items():
+            setattr(self, name, label)
```

Comment:
At this point, I prefer not to use ``setattr``, but literally assign it to ``self.labels``. ``label`` can also be added there. Something like this:
```python
self.labels = {"label": label, ** labels}
```

Reply:
Just to confirm this will mean that to update the labels during runtime the user will end up doing this:

```python
bar.label = "new label"
bar.labels["action"] = "new value"
```

If that's what we want then I'll make these changes.

Reply:
So inject `label` into `labels` in `__init__` and then provide a getter/setter `label` for backwards compatibility?

Reply:
oh I just processed the example you had in the first comment, yes we are on the same page